### ISSUE TITLE
Disable Mid-Turn Forfeiting

### ISSUE THREAD

**Issue by DaWoblefet** [Zarel/Pokemon-Showdown#4267]
In-game, a player is not allowed to forfeit a match until they have the ability to select the "Run" option during their turn. This includes situations such as these:
- a Pokemon fainting and player forfeiting before sending in their replacement Pokemon
- a Pokemon locked into Outrage / Shadow Force etc. in Singles and player forfeiting before they can select another option
- a Pokemon using U-turn / Volt Switch and player forfeiting before sending in their replacement Pokemon

While forfeiting mid-turn does offer convenience, it is a hindrance in team-locked best-of-three play (such as VGC), where forfeiting early can conceal information, such what the player's final Pokemon was.

**Comment by AustinXII** [comment]
If this is implemented it should only affect VGC imo

**Comment by SadisticMystic** [comment]
What if a player lets themselves run out of time in such situations as a proxy for forfeiting?

**Comment by Zarel** [comment]
In VGC, running out of time automatically selects your next pokemon/move, so it would be revealed anyway.

**Comment by Marty-D** [comment]
This only matters if "causing a communication error on purpose" doesn't incur harsh penalties. Obviously in a real VGC tournament you'd get disqualified or worse, but in your average "ladder" match that kind of thing happens all the time. On Battle Spot for example, they don't let you find a new match for some time.

The question is, are there plans to punish those who would "forfeit" early? It will always be possible to simply close your browser and keep that information to yourself anyway.

**Comment by Zarel** [comment]
It's relatively straightforward to simply let the game go on until the player's next choice, even if the player's left, if we want to do it that way.

**Comment by Marty-D** [comment]
My point was actually that the games don't do that; communication errors stop the battle in the same way that forfeiting currently does on PS. It's certainly an interesting problem.

**Comment by DaWoblefet** [comment]
Marty's correct in that PS's implementation would model that of a traditional disconnect. There are actually [very detailed guidelines](https://assets.pokemon.com//assets/cms2/pdf/play-pokemon/rules/play-pokemon-vg-rules-formats-and-penalty-guidelines-en.pdf) for VGC judges to follow if disconnects happen, from Section 3.4.5. This would assume that the disconnect is a double DC on both players' side however, which mid-turn forfeiting would not be in this instance. Section 5.4.1 says that a disconnect on the part of the player (e.g. turning the 3DS off) does only suggest to administer a game loss to the player. I'm reasonably confident that trying to pull off something like that in tournament, however, would result in a set loss or disqualification.

I'd be content with Zarel's approach of forcing moves that would occur at timeout (sending in next available Pokemon, forcing first moveslot), but that becomes a little more challenging if the player is unable to select their move that turn, e.g. both Pokemon are in the state of Outrage / are picking up the opponents with Sky Drop, and another action is required from the other player before the battle can progress. Marty brings another good point that this also is a complication to the average ladder match. Perhaps this issue should be shelved until a teamlocked best-of-three system is implemented?

**Comment by Zarel** [comment]
I can probably just implement cartridge-relevant selection mechanics in VGC...

**Comment by Zarel** [comment]
@DaWoblefet Did you end up getting clarification on that?

**Comment by Zarel** [comment]
Now, I assume something else happens on the ladder, though?

**Comment by Zarel** [comment]
GBU or Battle Spot or whatever they're calling it nowadays?

**Comment by DaWoblefet** [comment]
Yes, DCing on Battle Spot counts as a game loss. I know that multiple DCs can also get you disqualified from Global Link competitions, at least ones that are applicable to the Japanese Nationals qualification system (the International Challenges). Again, just to emphasize the point of this proposal, disabling mid-turn forfeits only really matter for team-locked best-of-three play, where you could potentially gain information in situations where you normally aren't going to be able to forfeit. Without support for best-of-three, I'm fine with this issue being shelved until then.

**Comment by Zarel** [comment]
I still can't close this, this is definitely something I'm going to need to deal with sooner or later. Best-of-three is nearly done: https://github.com/Zarel/Pokemon-Showdown/pull/4076

It's just waiting on the Preact rewrite, which is also nearly done.
